spock-core/src/main/java/org/spockframework/tempdir/TempDirConfiguration.java

```diff
@@ -0,0 +1,57 @@
+package org.spockframework.tempdir;
+
+import groovy.lang.Closure;
+import org.spockframework.runtime.extension.builtin.PreconditionContext;
+import org.spockframework.util.Beta;
+import spock.config.ConfigurationObject;
+
+import java.nio.file.Path;
+
+/**
+ *
+ * Configuration settings for the TempDir extension.
+ *
+ * <p>Example:
+ * <pre>
+ * tempdir {
+ *   baseDir false // default null, which means {@code System.getProperty("java.io.tmpdir")}
+ *   keep { os.windows } // default { false }, which is called with a delegate of type {@link PreconditionContext}
+ * }
+ * </pre>
+ * @author dqyuan
+ * @since 2.0
+ * @see PreconditionContext
+ */
+@Beta
+@ConfigurationObject("tempdir")
+public class TempDirConfiguration {
+  /**
+   * DefaultClosure always returns null, which means follow global configuration.
+   * And global configuration is false by default
+   */
+  public static class DefaultClosure extends Closure {
+    public DefaultClosure(Object owner, Object thisObject) {
+      super(owner, thisObject);
+    }
+
+    public Object doCall(Object args) {
+      return null;
```

Comment:
I would default to a system property `Boolean.getBoolean("spock.tempDir.keep")`

Reply:
Why? If a user wants it controllable from commandline he can easily program that in the configuation file. (Besides that I would probably change `Closure` to `boolean` now that it is no longer in the annotation)
I thought the preferable way for configuration is the configuration file except for things that must be set before like the custom path to the configuration file.
We recently removed the `spock.assertUnrollExpressions` system property and only left a configuration file option as replacement if you remember. :-)

Reply:
The difference for me would be that this would a temporary debugging change, i.e. I run into issues and want to keep my files to inspect them.  Sure the user could either set the closure to true, which runs the risk of getting committed accidentally, or he puts something like `Boolean.getBoolean("spock.tempDir.keep")` in there manually. If you work on only one project that is easy but if you have to put this in tens of projects this gets tedious.

Yes the preferred way to configure spock is the config file, but there are instances where the default of some keys can be changed via system property, e.g., `org.spockframework.report.log.ReportLogConfiguration#enabled`. This is what I meant here.

Using a plain `boolean` would probably be better.

Reply:
> If you work on only one project that is easy but if you have to put this in tens of projects this gets tedious
Well, you can set it in the spock configuration file in your spock home directory for all projects at once without the risk of committing. :-)
But yeah, I hear you.
Should the `baseDir` configuration then also react to a system property by default? (I mean besides `java.io.tempdir`, not instead of) @leonard84?